elastic/kibana
# [Infra UI] Add infrastructure metrics tables (#120377)

## Summary

Closes #120377 

This PR adds 3 table components to display metrics for 3 different node types (hosts, containers and pods).

Pod:
<img width="1270" alt="Screenshot 2022-03-15 at 12 39 31" src="https://user-images.githubusercontent.com/2564140/158370382-3c5d7531-00b6-4d89-a6e6-0c601686f2c2.png">

Host:
<img width="1275" alt="Screenshot 2022-03-15 at 12 39 48" src="https://user-images.githubusercontent.com/2564140/158370401-9c39abd6-c1d8-4c1c-b6e7-e584ab24c0df.png">

Container:
<img width="1267" alt="Screenshot 2022-03-15 at 12 39 41" src="https://user-images.githubusercontent.com/2564140/158370422-d1b3d3ea-5fac-4306-90d4-bb5d82d41167.png">

### How to test
For just the component visuals, please find it in [Storybook](https://ci-artifacts.kibana.dev/storybooks/pr-122352/5bd16697071f66fe6ee2641071ac29b512fa05dc/infra/index.html?path=/story/infra-node-metrics-tables-container--demo).

To test with real data:
I've been developing by dropping the component below into any Metrics UI page, like https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/pages/metrics/inventory_view/index.tsx for example.
Add the missing imports and add `<Demo />` some where in the page markup, like line 70 in the file above.

```js
function Demo() {
  const timerange = useMemo(
    () => ({
      from: 'now-30m',
      to: 'now',
    }),
    []
  );

  const filterClauseDsl = useMemo(
    () => ({
      bool: {
        should: [
          {
            match: {
              'host.name': 'gke-edge-oblt-pool-1-9a60016d-lgg9',
            },
          },
        ],
        minimum_should_match: 1,
      },
    }),
    []
  );

  const tableProps = useHostMetricsTable({
    timerange,
    filterClauseDsl,
  });
  return <HostMetricsTable {...tableProps} />;
}
```

### Open questions
1.  Do we even need any call out at all about the data being incomplete?
2. Is Container data also from Kubernetes or Docker metricset?
3. Does the node details page for Containers even work? The data seems empty.
4. Pod page also seems broken?
5. Can we tell if a pod/container is still running from the data we have?
6. Do we want any kind of telemetry added into these tables?
7. Can we avoid having APM load the source multiple times somehow since it can be expensive even if we fix https://github.com/elastic/kibana/issues/127616

### Follow up tasks:
Add tests: https://github.com/elastic/kibana/issues/127738
Migrate docs https://github.com/elastic/kibana/issues/127862
Add loading and error states: https://github.com/elastic/kibana/issues/127742

### Checklist
- [ ] This was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
